allennlp/data/tokenizers/character_tokenizer.py

```diff
@@ -17,34 +17,56 @@ class CharacterTokenizer(Tokenizer):
         If not ``None``, we will use this encoding to encode the string as bytes, and use the byte
         sequence as characters, instead of the unicode characters in the python string.  E.g., the
         character 'á' would be a single token if this option is ``None``, but it would be two
-        tokens if this option is set to ``"utf-8"``: there is one token for the accent and another
-        token for the 'a' character.  Note, though, that with a utf-8 encoding, the token for an
-        accented 'a' will be different than the token for an unaccented 'a' - hopefully your
-        embedding will decide that they are at least similar - while the token for the accent is
-        indeed shared across different accented vowels.
+        tokens if this option is set to ``"utf-8"``.
 
         If this is not ``None``, ``tokenize`` will return a ``List[int]`` instead of a
         ``List[str]``, and we will bypass the vocabulary in the ``TokenIndexer``.
     lowercase_characters : ``bool``, optional (default=``False``)
         If ``True``, we will lowercase all of the characters in the text before doing any other
         operation.  You probably do not want to do this, as character vocabularies are generally
         not very large to begin with, but it's an option if you really want it.
+    start_tokens : ``List[str]``, optional
+        If given, these tokens will be added to the beginning of every string we tokenize.  If
+        using byte encoding, this should actually be a ``List[int]``, not a ``List[str]``.
+    end_tokens : ``List[str]``, optional
+        If given, these tokens will be added to the end of every string we tokenize.  If using byte
+        encoding, this should actually be a ``List[int]``, not a ``List[str]``.
     """
-    def __init__(self, byte_encoding: str = None, lowercase_characters: bool = False) -> None:
-        self.byte_encoding = byte_encoding
-        self.lowercase_characters = lowercase_characters
+    def __init__(self,
+                 byte_encoding: str = None,
+                 lowercase_characters: bool = False,
+                 start_tokens: List[str] = None,
+                 end_tokens: List[str] = None) -> None:
+        self._byte_encoding = byte_encoding
+        self._lowercase_characters = lowercase_characters
+        self._start_tokens = start_tokens or []
+        self._start_tokens.reverse()
```

Comment:
What does this do?

Reply:
It's because of the `tokens.insert(0, start_token)` - you need to insert them in the reverse order so that they end up right.  I'll add a comment explaining this.